metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py

```diff
@@ -149,6 +150,10 @@ def get_query_result(self, query: str) -> RowIterator:
         return resp.result()
 
     def get_projects(self) -> List[BigqueryProject]:
+        def _should_retry(exc: BaseException) -> bool:
+            logger.debug(f"Exception reason: {str(exc)}. Type: {type(exc)}")
```

Comment:
add a comment here explaining what we previously tried and how we arrived at this solution